src/main.cpp

```diff
@@ -5713,6 +5713,9 @@ bool SendMessages(CNode* pto)
             pto->vAddrToSend.clear();
             if (!vAddr.empty())
                 pto->PushMessage(NetMsgType::ADDR, vAddr);
+            // we only send the big addr message once
+            if (pto->vAddrToSend.capacity() > 40)
+                pto->vAddrToSend = {};
```

Comment:
I just tested. Your old code indeed would not reduce capacity for me. *However* `pto->vAddrToSend = std::vector<CAddress>{}` does reduce it to 0. I wonder what the difference is between a type name being present before the {} or not.